### PR TITLE
APIGOV-22080 Update resource handling in deleting state

### PR DESCRIPTION
**Pull request by tsjohns9**
Update the handlers to not fetch a managed app when in deleting state

### CODE REVIEW

File: pkg/agent/handler/accessrequest.go [hunk 1]
```diff
@@ -60,10 +60,18 @@ func (h *accessRequestHandler) Handle(action proto.Event_Type, meta *proto.Event
 		return nil
 	}
 
-	if ok := shouldProcess(ar.Status.Level, ar.Metadata.State); !ok {
-		return nil
+	if ok := shouldProcessPending(ar.Status.Level, ar.Metadata.State); ok {
+		return h.onPending(ar)
+	}
+
+	if ok := shouldProcessDeleting(ar.Status.Level, ar.Metadata.State, len(ar.Finalizers)); ok {
```
Comment (tsjohns9) [inline pkg/agent/handler/accessrequest.go:67]: When a resource is in deleting state we end up creating another update event to remove the finalizer. So, I added a check to see if there are finalizers before deprovisioning.

**Comment by sbolosan** [comment]
@tsjohns9 @jcollins-axway , can we throw in some logging for the shouldNNNN funcs?

File: pkg/agent/handler/accessrequest.go [hunk 2]
```diff
@@ -74,43 +82,39 @@ func (h *accessRequestHandler) Handle(action proto.Event_Type, meta *proto.Event
 		return err
 	}
 
-	if action == proto.Event_DELETED {
```
Comment (tsjohns9) [inline pkg/agent/handler/accessrequest.go:77]: Do we need to do something on delete?

**Review by vivekschauhan — APPROVED** [review]

**Comment by sbolosan** [comment]
In the handlers... just maybe some trace logging. There was one section where it was calling funcs should"NNNN", and would be nice to know since it was being called back to back to back

File: pkg/agent/handler/accessrequest.go [hunk 3]
```diff
@@ -60,10 +58,20 @@ func (h *accessRequestHandler) Handle(action proto.Event_Type, meta *proto.Event
 		return nil
 	}
 
-	if ok := shouldProcess(ar.Status.Level, ar.Metadata.State); !ok {
-		return nil
+	if ok := shouldProcessPending(ar.Status.Level, ar.Metadata.State); ok {
+		log.Tracef("access request handler - processing resource in pending status")
```
[the remark this answers is not in the document]
Comment (sbolosan) [reply pkg/agent/handler/accessrequest.go:62]: 🤙

**Pull request merged**